src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items:T[], key: keyof T):GroupsMap<T> {
+  const seed:GroupsMap<T> = {};
+
+  return items.reduce((accumulator, item) => {
+    const groupName:string = String(item[key]);
```

Comment:
Generally in typescript we should rely as much as possible on type inference (typescript figuring out on its own the types) rather than explicit types, especially here where it obviously is a string. There may be an exception to this as some reviewers like to have explicit types, especially on return types of functions (I would say typing return types of functions is a thing that team of developers agree or not in their coding standards)